src/lib.rs

```diff
@@ -49,18 +49,10 @@ pub use zstd_safe::CLEVEL_DEFAULT as DEFAULT_COMPRESSION_LEVEL;
 #[doc(no_inline)]
 pub use stream::{decode_all, encode_all, Decoder, Encoder};
 
-/// Parse the result code
-///
-/// Returns the number of bytes written if the code represents success,
-/// or the error message otherwise.
-fn parse_code(code: usize) -> io::Result<usize> {
-    if zstd_safe::is_error(code) == 0 {
-        Ok(code)
-    } else {
-        let msg = zstd_safe::get_error_name(code);
-        let error = io::Error::new(io::ErrorKind::Other, msg.to_string());
-        Err(error)
-    }
+/// Returns the error message as io::Error based on error_code.
+fn map_error_code(code: u32) -> io::Error {
```

Comment:
Actually most (all?) error codes won't fit in `u32` since they are defined as negative `size_t` values (so they'll be very close to usize::max_value`.

Reply:
`u32` was mistaken, because I thought `ZSTD_isError` returns error code. Now it is fixed.